sources/api/certdog/src/main.rs

```diff
@@ -54,34 +57,34 @@ struct CertBundle {
 
 /// Query the API for the certificate bundles, returns a tuple with trusted
 /// and distrusted PEM certificates
-async fn get_certificate_bundles<P>(socket_path: P) -> Result<CertBundle>
+async fn get_certificate_bundles<P>(config_path: P) -> Result<CertBundle>
 where
     P: AsRef<Path>,
 {
-    debug!("Querying the API for settings");
-
-    let method = "GET";
-    let uri = constants::API_SETTINGS_URI;
-    let (_code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
-        .await
-        .context(error::APIRequestSnafu { method, uri })?;
+    debug!("Loading certdog configuration");
+    if !config_path.as_ref().exists() {
```

Comment:
nit: can remove some duplication by doing the `as_ref()` call once to get a `Path`:
```suggestion
    debug!("Loading certdog configuration");
    let config_path = config_path.as_ref();
    if !config_path.exists() {
```